### ISSUE TITLE
Change icon for unknown profiles?

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper-maps#376]
In https://graphhopper.slack.com/archives/DCYDKS5RB/p1710765026728319 we added a question mark icon for unknown profiles. The UI might look like this for custom profile names:

![image](https://github.com/graphhopper/graphhopper-maps/assets/17603532/f1f7a04c-5148-4673-8986-15c97d4e059a)

This looks somewhat unintentional and buggy. The good thing is that we get to see the profile name when we hover the icons. But maybe we can use another icon in this case?

**Comment by SachsenspieltCoding** [comment]
@easbar Yes, I see your point. I actually thought of that when I added it, which is why I mentioned it in the PR.

How do you think it could be handled better? Just replace the icon with something more meaningful, or can you think of a completely different approach?

**Comment by easbar** [comment]
Maybe some icon that is 1) less confused-looking and 2) something that allows the profiles to be distinguished from each other. For example we could just use icons showing numbers 1, 2, 3, ... ? Or a car icon with a little number next to it? Then again they aren't necessarily cars so maybe something more generic? But honestly I don't have a good suggestion atm, so maybe just numbers will do for now?

**Comment by SachsenspieltCoding** [comment]
It would be an idea. But aren't numbers even more confusing? I mean, when there are a bunch of contextless numbers in the profiles area, it may lead to even more confusion what a "3" or a "5" means for instance.

The best option I can think of is to allow some sort of custom icons  respectively the general option to set icons for custom profiles.

**Comment by easbar** [comment]
Maybe it is just me, but I think numbers 1,2,3,... would be less confusing (and seem less buggy) for me than three question marks. To figure out which one is which we still need to hover the icon of course.

**Comment by SachsenspieltCoding** [comment]
Perhaps @karussell could give his opinion on this?

While I'm still not 100% sure about the number icons, I have no problem implementing this in a PR.
Also, I would prefer an option to set an icon manually, but this would require a new field in the `RoutingProfile` interface, which is implemented by the `graphhopper.d.ts` file.

https://github.com/graphhopper/graphhopper-maps/blob/master/src/api/graphhopper.d.ts#L59-L61

Is it possible to add it there, or should I then add a second `.d.ts` file that implements this interface?

**Comment by easbar** [comment]
> Also, I would prefer an option to set an icon manually, but this would require a new field in the RoutingProfile interface, 

Can we not just do this using the config file?

**Comment by SachsenspieltCoding** [comment]
Yes, that is the plan. But since the config is more or less typed through the TypeScript interfaces, this would also require a change in the type files. There may be a way around this, I haven't fully analysed it yet.

**Comment by easbar** [comment]
The config is typed in custom.d.ts and sure you can modify it accordingly when you are adding the icon option

**Comment by SachsenspieltCoding** [comment]
Oh yes, thanks for the hint!

**Comment by SachsenspieltCoding** [comment]
![grafik](https://github.com/graphhopper/graphhopper-maps/assets/59836848/bd5c3f49-6ccb-4bea-aa48-df5343546848)

This is what you thought of, right?

**Comment by easbar** [comment]
Yes, that looks good?! Definitely better than the question marks if you ask me. Maybe just start counting the first unknown profile as "1"(?)

**Comment by SachsenspieltCoding** [comment]
Yeah, I also thought of this. It would require more code, so I just decided for the easier option first.
I'll try my best.

**Comment by SachsenspieltCoding** [comment]
![grafik](https://github.com/graphhopper/graphhopper-maps/assets/59836848/b3ed6e81-a46d-445f-b0ab-188212c11288)

OK, so now only those profiles without an icon will be counted.
You can now also set an icon using the "icon" field in the config.

I will probably open a PR today, until then you can see the changes here
https://github.com/SachsenspieltCoding/graphhopper-maps/commit/5b4cab4f91bace0d9072777c4dc99bc8f964488d

**Comment by SachsenspieltCoding** [comment]
Sounds possible. But as it would require more sorting, and more intelligent counting, I'll one or two days extra time

**Comment by SachsenspieltCoding** [comment]
What do you think? I've loaded the default profiles from the GraphHopper API to test it out. The styling is work in progress.

![grafik](https://github.com/graphhopper/graphhopper-maps/assets/59836848/859f7ac8-7642-4db4-9437-4cb72f8cfe50)

---

There is one edge case I came across that I would like to discuss here. Let's say I have the profiles
`car, car_veryfast, slowvehicle`.
By default it would look like this:
![grafik](https://github.com/graphhopper/graphhopper-maps/assets/59836848/068041b6-2529-46a5-82cc-eeee18fc38ec)
But what if I manually change the `slowvehicle` icon to a car icon? Well, we can do two things
1) Ignore counting and set the icon normally
2) Behave as if it were named `car_slowvehicle`, so that it also has the number in the top right corner.

The current implementation is 1), which I think has the most potential for customisation.
![grafik](https://github.com/graphhopper/graphhopper-maps/assets/59836848/c575f76d-118c-46c0-aea5-d2f400484ff6)

**Comment by SachsenspieltCoding** [comment]
For the protocol: The option to set icons via an attribute in the profile config **will not be supported**. Instead, the focus is on the prefixes, which technically allow custom profiles in a way that is enough for the most users.
